lib/power_action_utils.pm

```diff
@@ -220,30 +221,29 @@ sub power_action {
             }
         }
     }
+    my $soft_fail_data;
     # Shutdown takes longer than 60 seconds on SLE 15
     my $shutdown_timeout = 60;
     if (is_sle('15+') && check_var('DESKTOP', 'gnome') && ($action eq 'poweroff')) {
-        record_soft_failure('bsc#1055462');
-        $shutdown_timeout *= 3;
+        $soft_fail_data = {bugref => 'bsc#1055462', soft_timeout => 60, timeout => 180};
     }
     # The timeout is increased as shutdown takes longer on Live CD
     if (get_var('LIVECD')) {
-        $shutdown_timeout *= 4;
+        $soft_fail_data = {bugref => 'bsc#1096241', soft_timeout => 60, timeout => 300};
     }
     if (get_var("OFW") && check_var('DISTRI', 'opensuse') && check_var('DESKTOP', 'gnome') && get_var('PUBLISH_HDD_1')) {
-        $shutdown_timeout *= 3;
-        record_soft_failure("boo#1057637 shutdown_timeout increased to $shutdown_timeout (s) expecting to complete.");
+        $soft_fail_data = {bugref => 'bsc#1057637', soft_timeout => 60, timeout => 180};
     }
     # no need to redefine the system when we boot from an existing qcow image
     # Do not redefine if autoyast or s390 zKVM reboot, as did initial reboot already
     if (check_var('VIRSH_VMM_FAMILY', 'kvm')
         || check_var('VIRSH_VMM_FAMILY', 'xen')
         || (get_var('S390_ZKVM') && !get_var('BOOT_HDD_IMAGE') && !get_var('AUTOYAST') && $args{first_reboot}))
     {
-        assert_shutdown_and_restore_system($action, $shutdown_timeout);
+        assert_shutdown_and_restore_system($action);
     }
     else {
-        assert_shutdown($shutdown_timeout) if $action eq 'poweroff';
+        assert_shutdown_with_soft_timeout($soft_fail_data) if ($action eq 'poweroff');
```

Comment:
`$soft_fail_data` hash can be undef here, which will cause failure in the method below.

Reply:
@rwx788, there is a verification run in the PR comment body, that covers this case: http://10.160.65.138/tests/228

Also, I've re-checked the same again. I've  left the following and executed the test (commented all the lines, where the $soft_fail_data hash was defined):

```perl
my $soft_fail_data;
else {
        assert_shutdown_with_soft_timeout($soft_fail_data) if ($action eq 'poweroff');
...
```

And the test is passed: http://oorlov-vm.qa.suse.de/tests/116

I do not understand why should it fail? If $soft_fail_data is not specified, then `$args->{timeout} //= 60;` in `assert_shutdown_with_soft_timeout` will be called.

Please correct me if you see something potentially wrong.